contracts/nft/TributeERC721.sol

```diff
@@ -0,0 +1,218 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.4;
+
+import "@openzeppelin/contracts-upgradeable/token/ERC721/ERC721Upgradeable.sol";
+import "@openzeppelin/contracts-upgradeable/access/OwnableUpgradeable.sol";
+import "@openzeppelin/contracts-upgradeable/proxy/utils/Initializable.sol";
+import "@openzeppelin/contracts-upgradeable/proxy/utils/UUPSUpgradeable.sol";
+import "@openzeppelin/contracts-upgradeable/utils/CountersUpgradeable.sol";
+import "@openzeppelin/contracts-upgradeable/utils/StringsUpgradeable.sol";
+import "../core/DaoRegistry.sol";
+import "../utils/Signatures.sol";
+
+contract TributeERC721 is
+    Initializable,
+    ERC721Upgradeable,
+    OwnableUpgradeable,
+    UUPSUpgradeable,
+    Signatures
+{
+    struct Checkpoint {
+        // A checkpoint for marking number of votes from a given block
+        uint96 fromBlock;
+        uint160 amount;
+    }
+
+    using CountersUpgradeable for CountersUpgradeable.Counter;
+    using StringsUpgradeable for uint256;
+
+    CountersUpgradeable.Counter private _tokenIdCounter;
```

Comment:
I don't think you need that anymore. 
You are using nonce now. We can simply say that `require(nonce < collection size)` and remove this variable